gateleen-core/src/main/java/org/swisspush/gateleen/core/util/SlicedLoop.java

```diff
@@ -0,0 +1,208 @@
+package org.swisspush.gateleen.core.util;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClientRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static java.lang.System.nanoTime;
+
+public class SlicedLoop<T> {
```

Comment:
Propose to describe on class level what this is.

Reply:
Good point. While trying to describe it, I found that the class by fact did multiple things at the same time. I now split off another class and now it was easier to describe what they're doing.